### PR TITLE
Sliced shape has tests, and can deal with np.newaxis.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

**Comment by pelson** [comment]
If nothing else, the functions are of use here - I'd happily strip this back to just tests if necessary.

File: biggus/__init__.py [hunk 1]
```diff
@@ -2236,34 +2235,39 @@ def sub(a, b):
...
             # Numpy boolean indexing.
             sliced_shape.append(__builtin__.sum(key))
         elif isinstance(key, (tuple, np.ndarray)):
             sliced_shape.append(len(key))
+        elif key is np.newaxis:
+            shape_dim -= 1
+            sliced_shape.append(1)
         else:
-            sliced_shape.append(size)
+            sliced_shape.append(shape[shape_dim])
```
Comment (rhattersley) [inline biggus/__init__.py]: Is this line ever used?

Comment (pelson) [inline biggus/__init__.py]: A fine question. Exchanging it for an exception would be a good test...

**Comment by pelson** [comment]
I've rebased (to make use of #121), and removed the catch-all else statement.

**Comment by pelson** [comment]
Ping @rhattersley & @bjlittle - I have a PR waiting in the wings which will really benefit from this - it is an alternative implementation of #116.

**Comment by rhattersley** [comment]
Thanks @pelson

**Pull request merged**